packages/icons/src/svgIconContainer.tsx

```diff
@@ -42,7 +42,7 @@ export type SVGIconContainerProps<T extends Element> = Omit<SVGIconProps<T>, "ch
  * @see https://stackoverflow.com/a/73795494/7406866
  */
 export interface SVGIconContainerComponent extends React.FC<SVGIconContainerProps<Element>> {
-    <T extends Element = Element>(props: SVGIconContainerProps<T>): React.ReactElement | null;
```

Comment:
Similar to [type compile issue with Icon](https://github.com/palantir/blueprint/pull/7142#discussion_r1915691062), this fix compiles in React 18, but not 16.

See: https://github.com/palantir/blueprint/compare/gdouglas/test-SVGIconContainer-compile-break